src/components/CommandBar/index.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useCallback } from 'react';
+
+import { useHotkeys } from 'react-hotkeys-hook';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+
+import CommandBarBody from './CommandBarBody';
+
+import Modal from 'src/components/dls/Modal/Modal';
+import { selectCommandBarIsOpen, setIsOpen, toggleIsOpen } from 'src/redux/slices/commandBar';
+
+const CommandBar: React.FC = () => {
+  const dispatch = useDispatch();
+  const isOpen = useSelector(selectCommandBarIsOpen, shallowEqual);
+  const toggleShowCommandBar = useCallback(() => {
+    dispatch({ type: toggleIsOpen.type });
+  }, [dispatch]);
+  const closeCommandBar = useCallback(() => {
+    dispatch({ type: setIsOpen.type, payload: false });
+  }, [dispatch]);
+  useHotkeys('cmd+k', toggleShowCommandBar, [dispatch]);
+  useHotkeys('Escape', closeCommandBar, { enabled: isOpen, enableOnTags: ['INPUT'] }, [dispatch]);
+
+  return (
+    <Modal isOpen={isOpen} onClickOutside={closeCommandBar}>
+      <CommandBarBody />
```

Comment:
can we render this conditionally only when the modal is open (not sure if that's handled as part of the modal code). This way we can make our dom footprint smaller.

Reply:
It's handled as part of the Modal. Anything that is inside the `Modal` is ever mounted in the DOM only when the `Modal` is open.